src/main/java/thePackmaster/powers/anomalypack/ThoughtweavingPower.java

```diff
@@ -0,0 +1,104 @@
+package thePackmaster.powers.anomalypack;
+
+import com.megacrit.cardcrawl.actions.common.DrawCardAction;
+import com.megacrit.cardcrawl.actions.utility.UseCardAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.cards.CardGroup;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.core.CardCrawlGame;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.localization.PowerStrings;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import thePackmaster.SpireAnniversary5Mod;
+import thePackmaster.cards.AbstractPackmasterCard;
+import thePackmaster.packs.AbstractCardPack;
+import thePackmaster.powers.AbstractPackmasterPower;
+import java.util.Iterator;
+
+import static thePackmaster.util.Wiz.att;
+
+public class ThoughtweavingPower extends AbstractPackmasterPower {
+    public static final String POWER_ID = SpireAnniversary5Mod.makeID("ThoughtweavingPower");
+    private static final PowerStrings powerStrings = CardCrawlGame.languagePack.getPowerStrings(POWER_ID);
+    public static final String NAME = powerStrings.NAME;
+    public static final String[] DESCRIPTIONS = powerStrings.DESCRIPTIONS;
+    private static final Logger logger = LogManager.getLogger("Packmaster");
+
+    public ThoughtweavingPower(AbstractCreature owner, int amount) {
+        super(POWER_ID, NAME, PowerType.BUFF,false, owner, amount);
+    }
+
+    public void onUseCard(AbstractCard tmpCard, UseCardAction action) {
+        //logger.info("Check 0");
+        if (tmpCard instanceof AbstractPackmasterCard) {
+            //logger.info("Check 1");
+            AbstractPackmasterCard card = (AbstractPackmasterCard) tmpCard;
+            if (card.type == AbstractCard.CardType.ATTACK || card.type == AbstractCard.CardType.SKILL) {
+                this.flash();
+                //logger.info("Check 2");
+                if (AbstractDungeon.player.hasPower("No Draw")) {
```

Comment:
Please use NowDrawPower.ID or POWER_ID whatever it is. Don#t pass string literals